discovery-provider/graphql/src/datasources/db.ts

```diff
@@ -0,0 +1,25 @@
+import DataLoader from 'dataloader'
+import type { Knex } from 'knex'
+
+export const DiscoveryDB = (dbConnection: Knex) => {
+  const blockConfirmationDataLoader = new DataLoader(
+    async (blockNumbers: readonly number[]) => {
+      const res = await dbConnection
+        .table('blocks')
+        .select('number')
+        .where('is_current', true)
+
+      if (res.length !== 1) {
+        throw new Error('Expected SINGLE row in blocks marked as current')
+      }
```

Comment:
you can add a `.first()` here and avoid this check as well as the throwing
https://knexjs.org/guide/query-builder.html#first